### PR TITLE
FileNotFoundException thrown by Web Authenticator in UWP. Fixed #1357

### LINKED ISSUE #1357: [Bug] FileNotFoundException is thrown when internet is turned off and user closes WAB dialog in uwp
### Description
When internet is turned off and user closes `WebAuthenticationBroker` dialog, `FileNotFoundException` is thrown.

### Steps to Reproduce
1.  Download Xamarin Essentials Samples
2.  Turn off internet.
3. Try to use `Web Authenticator` in UWP.

### Expected Behavior
When the dialog is closed, `WebAuthenticationStatus.UserCancel` is returned and then `TaskCanceledException` is thrown.

### Actual Behavior
FileNotFoundException is thrown

### Basic Information
- Version with issue: All
- Last known good version: None
- IDE: Visual Studio Community 2019
- Platform Target Frameworks: Xamarin Essentials Samples

### Screenshots
Immagine1
Immagine2

### PR DESCRIPTION
**Pull request by IlGalvo**
When internet is turned off and user closes the WAB dialog in uwp, FileNotFoundException is thrown.

### Description of Change ###
Describe your changes here. 

### Bugs Fixed ###
- Related to issue #1357 

### API Changes ###
None

### Behavioral Changes ###
Like android and ios do, when dialog is closed, a TaskCanceledException is thrown.

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of master at time of PR
- [ ] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

**Comment by IlGalvo** [comment]
No, it's not documented at all: https://docs.microsoft.com/en-us/uwp/api/windows.security.authentication.web.webauthenticationbroker.authenticateasync?view=winrt-19041.

I think it is an error resulting from the base implementation in c/c++ of the `WebAuthenticationBroker.AuthenticateAsync` that could probably use some sort of cache stored in files.

**Comment by mattleibow** [comment]
> No, it's not documented at all

Total boo. I was having a look at other things, and it seems like this happens often without network and things like that.
Why don't they just catch or wrap it?

File: Xamarin.Essentials/WebAuthenticator/WebAuthenticator.uwp.cs [hunk 1]
```diff
@@ -17,20 +19,27 @@ static async Task<WebAuthenticatorResult> PlatformAuthenticateAsync(Uri url, Uri
             if (!IsUriProtocolDeclared(callbackUrl.Scheme))
                 throw new InvalidOperationException($"You need to declare the windows.protocol usage of the protocol/scheme `{callbackUrl.Scheme}` in your AppxManifest.xml file");
 
-            var r = await WebAuthenticationBroker.AuthenticateAsync(WebAuthenticationOptions.None, url, callbackUrl);
```
[the remark this answers is not in the document]
Comment (IlGalvo) [reply Xamarin.Essentials/WebAuthenticator/WebAuthenticator.uwp.cs:20]: Yes, the line that fails is `WebAuthenticationBroker.AuthenticateAsync`, so there are not valid response available.

**Comment by IlGalvo** [comment]
Dunno, maybe is a lack, maybe is by design.

Take a look of `WebAuthenticationStatus` https://docs.microsoft.com/en-us/uwp/api/windows.security.authentication.web.webauthenticationstatus?view=winrt-19041: it wraps only the cases we are already handling.

Now, in the PR i made, I try-catched `FileNotFoundException` in `WebAuthenticationBroker.AuthenticateAsync`, and, to align the android/ios behavior when the the dialog is closed, I throw `TaskCanceledException`.

Tell me if you think.

**Comment by mattleibow** [comment]
@redth what is the correct branch for this bug? 1.6 or develop?

**Comment by mattleibow** [comment]
@IlGalvo thanks for this PR! Great to see it get into the library!

**Pull request merged**